x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/QueryList.java

```diff
@@ -99,7 +98,7 @@ private static IntFunction<Object> blockToJavaObject(Block block) {
             }
             case NULL -> offset -> null;
             case DOC -> throw new EsqlUnsupportedOperationException("can't read values from [doc] block");
-            case UNKNOWN -> throw new EsqlIllegalArgumentException("can't read values from [" + block + "]");
```

Comment:
UOE is used for methods not implemented - EIAE should be used instead both for UNKNOWN and DOC.